draft-ietf-dnsop-ns-revalidation.mkd

```diff
@@ -175,31 +177,33 @@ Attacks exploiting lack of this revalidation have been described in {{GHOST1}},
-    Successive queries directed to the same zone will be directed to the authoritative values for the names of the NS RRset in the referral response.
+When a resolver detects that the child's apex NS RRset contains different nameservers than the non-authoritative version at the parent side of the zone cut, it MAY report the mismatch using DNS Error Reporting {{RFC9567}} on the Report-Channel for the child zone, as well as on the Report-Channel for the parent zone, with an extended DNS error code of TBD {{IANA}}.
 
--   The names from the NS RRset from a validating NS query may differ from the names in NS RRset in the referral response.
-    Outstanding validation queries for "glue" that do not match names in the authoritative NS RRset be discarded, or they may be left running to completion.
-    Their result will no longer be used in queries for the zone.
-    Outstanding validation queries for "glue" that do match names in the authoritative NS RRset must be left running to completion.
-    They do not need to be re-queried after reception of the authoritative NS RRset (see [](#upgrade-addresses)).
+Additional validation queries for the "glue" resource records of referral responses (if not already authoritatively present in cache) may be sent with the validation query for the NS RRset as well.
+Positive responses will be cached authoritatively and replace the non authoritative "glue" entries.
+Successive queries directed to the same zone will be directed to the authoritative values for the names of the NS RRset in the referral response.
 
--   Resolvers may choose to delay the response to the triggering query until both the triggering query and the validation query have been answered.
-    In practice, we expect many implementations may answer the triggering query in advance of the validation query for performance reasons.
-    An additional reason is that there are unfortunately a number of nameservers in the field that (incorrectly) fail to properly answer explicit queries for zone apex NS records, and thus the revalidation logic may need to be applied lazily and opportunistically to deal with them.
-    In cases where the delegated nameservers respond incorrectly to an NS query, the resolver should abandon this algorithm for the zone in question and fall back to using only the information from the parent's referral response.
+The names from the NS RRset from a validating NS query may differ from the names in NS RRset in the referral response.
+Outstanding validation queries for "glue" that do not match names in the authoritative NS RRset be discarded, or they may be left running to completion.
+Their result will no longer be used in queries for the zone.
+Outstanding validation queries for "glue" that do match names in the authoritative NS RRset must be left running to completion.
+They do not need to be re-queried after reception of the authoritative NS RRset (see [](#upgrade-addresses)).
```

Comment:
s/reception/receipt/

Reply:
Thanks. Resolved in [a8303ea](https://github.com/shuque/ns-revalidation/pull/26/commits/a8303ea7b433fa5bf6dd462d36a576d19d926ec8)